templates/bootstrap/Makefile.j2

```diff
@@ -0,0 +1,43 @@
+COMMIT?=HEAD
+WORKDIR?=.
+VENV?=$(WORKDIR)/.venv
+
+all: lint test
+
+clean:
+	rm -rf $(VENV)
+
+setup-venv:
+	python -m venv $(VENV)
```

Comment:
Can we ensure that this will be a python 3 venv?

Reply:
Usually, i let the user handle the venv stuff outside of the Makefile.
That helps to reuse the Makefile in CI, where you install different versions of python by other means.

Reply:
I agree. Putting venv stuff in the Makefile is not optimal. The problem I am trying to solve is that different plugins will have different sets of packages for linting/testing/etc that may conflict. And right now in Pulp dev environments we just use a single virtualenv called 'pulp'. Do you know of a better solution?

Reply:
Sort of. We can put the parts for linting in another section of the plugin template that can be updated (i.e. not bootstrap).
And reunify them throughout the project.

Reply:
How do we deal with the situation where a plugins are requiring flake8 plugins that other plugins don’t use? Once a flake8 plugin gets installed in the pulp virtualenv, it runs against all projects.

Reply:
My one thought is to make `setup-venv` optional and controlled by a variable.

Reply:
> How do we deal with the situation where a plugins is requiring flake8 plugins that other plugins don’t use? Once a flake8 plugin gets installed in the pulp virtualenv, it runs against all projects.

That is exactly the problem, why running flake in the dev environment is a pain today. I'd like all plugins to require (and install) the same set of flake plugins.

Reply:
maybe we could add ignore rules that ingore all codes from a plugin with a wildcard?

Reply:
> That is exactly the problem, why running flake in the dev environment is a pain today. I'd like all plugins to require (and install) the same set of flake plugins.

I feel differently: we should encourage plugins to use the same code styles, etc but not force it upon them. If a plugin wants to use say flake8-mypy they should be free to do so. How would we stop them from doing so?

Reply:
I like the "play nice" ansatz. So we should try to solve the root case being that flake plugins are activated only by installing them.

Maybe this helps: https://stackoverflow.com/questions/41860123/how-to-ignore-installed-flake8-plugin-easily-for-one-time

Reply:
I think having a command to setup a venv in the Makefile and then use it if it exists is a good solution in that it shouldn't affect users who don't want to use a venv. That said I'm not opposed to also having flake8 ignore certain plugins when it's run (or better yet we should probably use `--select`) if we want to also add that.